backend/kernelCI_app/management/commands/process_pending_aggregations.py

```diff
@@ -0,0 +1,394 @@
+
+            if not pending_tests_batch:
+                self.stdout.write("No pending tests found, exiting batch")
+                return 0
+
+            last_processed_test_id = pending_tests_batch[-1].test_id
+
+            pending_build_ids = {pt.build_id for pt in pending_tests_batch}
+            found_builds = Builds.objects.in_bulk(list(pending_build_ids))
+
+            if not found_builds:
+                self.stdout.write(
+                    self.style.WARNING(
+                        "No builds found for pending tests, skipping batch"
+                    )
+                )
+                continue
+
+            required_checkout_ids = {b.checkout_id for b in found_builds.values()}
+            found_checkouts = Checkouts.objects.in_bulk(list(required_checkout_ids))
+
+            ready_tests = []
+            ready_builds = {}
+            ready_checkouts = {}
+            skipped_no_build = 0
+            skipped_no_checkout = 0
+
+            for pt in pending_tests_batch:
+                build = found_builds.get(pt.build_id)
+                if not build:
+                    skipped_no_build += 1
+                    continue
+
+                checkout = found_checkouts.get(build.checkout_id)
+                if not checkout:
+                    skipped_no_checkout += 1
+                    continue
+
+                ready_tests.append(pt)
+                ready_builds[build.id] = build
+                ready_checkouts[checkout.id] = checkout
+
+            if not ready_tests:
+                continue
+
+            self._process_ready_tests(
+                ready_tests,
+                ready_builds,
+                ready_checkouts,
+                skipped_no_build,
+                skipped_no_checkout,
+            )
+
+    def _process_ready_tests(
+        self,
+        ready_tests,
+        ready_builds,
+        ready_checkouts,
+        skipped_no_build,
+        skipped_no_checkout,
+    ) -> int:
+        with transaction.atomic():
+            hardware_status_data, new_processed_entries = aggregate_hardware_status(
+                ready_tests, ready_builds, ready_checkouts
+            )
+
+            if hardware_status_data:
+                values = [
+                    (
+                        data["checkout_id"],
+                        data["origin"],
+                        data["platform"],
+                        data["compatibles"],
+                        data["start_time"],
+                        data["build_pass"],
+                        data["build_failed"],
+                        data["build_inc"],
+                        data["boot_pass"],
+                        data["boot_failed"],
+                        data["boot_inc"],
+                        data["test_pass"],
+                        data["test_failed"],
+                        data["test_inc"],
+                    )
+                    for data in hardware_status_data.values()
```

Comment:
`values_list("checkout_id", "origin"...)` will return all items as an array in the correct order, much cheaper and no need to loop again